File: hrms/src/main/java/com/kaya/hrms/business/abstracts/JobAdvertisementService.java
package com.kaya.hrms.business.abstracts;

import java.util.List;

import com.kaya.hrms.core.utilities.results.DataResult;
import com.kaya.hrms.core.utilities.results.Result;
import com.kaya.hrms.entities.Dtos.JobAdvertisementAddDto;
import com.kaya.hrms.entities.Dtos.JobAdvertisementWithCompanyDto;
import com.kaya.hrms.entities.concretes.JobAdvertisement;

public interface JobAdvertisementService {
	
	DataResult<List<JobAdvertisement>> getAll();
	
	DataResult<List<JobAdvertisement>> getByEnable();
	
	DataResult<List<JobAdvertisement>> getByEnable(int pageNo, int pageSize);

	DataResult<List<JobAdvertisement>> getByEnableSortedAsc();

	DataResult<List<JobAdvertisement>> getByEnableSortedDesc();
	
	DataResult<List<JobAdvertisementWithCompanyDto>> getJobAdvertisementWithCompany();
	
	DataResult<List<JobAdvertisementWithCompanyDto>> getJobAdvertisementWithCompanyOrderByCreatedAt();

	DataResult<List<JobAdvertisement>> getByEnableTrueAndCompanyName(String companyName);
	

	Result add(JobAdvertisementAddDto jobAdvertisementAddDto);
	
	Result update(int jobAdvertisemenetId, JobAdvertisement jobAdvertisemenet);


}

File: hrmsfrontend/src/services/jobAdvertisementService.js
import axios from "axios";

export default class JobAdvertisementService {
  constructor() {
    this.apiUrl = "http://localhost:8080/api/jobAdvertisements/";
  }

  getJobAdvertisements() {
    return axios.get(`${this.apiUrl}getAll`);
  }

  getJobAdvertisementsByEnable() {
    return axios.get(`${this.apiUrl}getByEnable`);
  }

  getJobAdvertisementsByEnableAndCompanyName(companyName) {
    return axios.get(`${this.apiUrl}getByEnableAndCompany_companyName?companyName=${companyName}`);
  }

  getJobAdvertisementsByEnablePage(pageNo, pageSize) {
    return axios.get(`${this.apiUrl}getByEnablePage?pageNo=${pageNo}&pageSize=${pageSize}`);
  }

  getJobAdvertisementsByEnableSortedAsc() {
    return axios.get(`${this.apiUrl}getByEnableSortedAsc`);
  }

  getJobAdvertisementsByEnableSortedDesc() {
    return axios.get(`${this.apiUrl}getByEnableSortedDesc`);
  }

  getJobAdvertisementsWithCompany() {
    return axios.get(`${this.apiUrl}getJobAdvertisementWithCompany`);
  }

  getJobAdvertisementsWithCompanyOrderByCreatedAt() {
    return axios.get(`${this.apiUrl}getJobAdvertisementWithCompanyOrderByCreatedAt`);
  }

  add(jobAdvertisement) {
    axios.post(`${this.apiUrl}add`, jobAdvertisement);
  }

  update(jobAdvertisementId, jobAdvertisement) {
    axios.put(`${this.apiUrl}update?jobAdvertisemenetId=${jobAdvertisementId}`, jobAdvertisement)
  }

  confirmation(jobAdvertisementId, confirm) {
    axios.put(`${this.apiUrl}confirmation?confirm=${confirm}&jobAdvertisemenetId=${jobAdvertisementId}`)
  }

}
